<launch>
  <arg name="output_screen" default="true"/>
  <arg name="image_topic" default="image" />
  <arg name="image_depth_topic" default="image_depth" />
  <arg name="camera_info_topic" default="cam_info" />
  <arg name="debug_view" default="true"/>
  <arg name="marker_debug" default="true"/>
  <arg name="poly_array" default="polygon_array" />
  <arg name="frame_id" default="target_object"/>
  <arg name="camera" default="rs_d435"/>


  <node pkg="nodelet" type="nodelet" name="cluttered_brick_detection_nodelet" args="manager" output="screen" if="$(arg output_screen)"/>
  
  <node pkg="nodelet" type="nodelet" name="cluttered_brick_detection"
        args="load aerial_robot_perception/ClutteredBrickDetection cluttered_brick_detection_nodelet" output="screen">

    <remap from="cam_info" to="/rs_d435/aligned_depth_to_color/camera_info" />
    <remap from="polygon_array" to="/multi_plane_estimate/output_refined_polygon" />
    <remap from="depth_image" to="/rs_d435/aligned_depth_to_color/image_raw" />
    
    <param name="debug_view" value="$(arg debug_view)"/>
    <param name="frame_id" value="$(arg frame_id)"/>
    <param name="marker_debug" value="$(arg marker_debug)"/>
  </node>

</launch>
